R/indexing.R

```diff
@@ -53,6 +53,55 @@
 # - G[1:3,2,eid=TRUE]
 #               create an edge sequence
 
+get_adjacency_submatrix <- function(x, i, j, attr = NULL, sparse = TRUE) {
+  # If i or j is NULL, assume all nodes
+  # if not NULL make sure to handle duplicates correctly
+  if (missing(i)) {
+    i <- seq_len(vcount(x))
+  }
+  if (missing(j)) {
+    j <- seq_len(vcount(x))
+  }
```

Comment:
Is this worth the effort?

```suggestion
  if (missing(i)) {
    i_seq <- seq_len(vcount(x))
    has_i <- FALSE
  } else {
    i_seq <- i
    has_i <- TRUE
  }
  has_j <- missing(j)
```

and then only using `i_seq`, `has_i` and `has_j` to conditionally call `match()` and `%in%` ?

Reply:
> Is this worth the effort?
> 
> and then only using `i_seq`, `has_i` and `has_j` to conditionally call `match()` and `%in%` ?

Seems so. We do gain some time and I think it is enough to justify the implementation